featuretools/tests/primitive_tests/test_install_primitives.py

```diff
@@ -0,0 +1,126 @@
+    installation_dir = get_installation_dir()
+    files = list_primitive_files(installation_dir)
+    # then delete to clean up
+    for f in files:
+        os.unlink(f)
+
+
+def do_installation(install_path):
+    featuretools.primitives.install.install_primitives(install_path, prompt=False)
+
+    # must reload submodule for it to work
+    reload(featuretools.primitives.installed)
+    from featuretools.primitives.installed import CustomMax, CustomSum, CustomMean  # noqa: F401
+
+    installation_dir = get_installation_dir()
+    files = list_primitive_files(installation_dir)
+    custom_max_file = os.path.join(installation_dir, "custom_max.py")
+    custom_mean_file = os.path.join(installation_dir, "custom_mean.py")
+    custom_sum_file = os.path.join(installation_dir, "custom_sum.py")
+    assert set(files) == {custom_max_file, custom_mean_file, custom_sum_file}
+
+
+def test_install_primitives_directory(primitives_to_install_dir):
+    setup_installation()
+    do_installation(primitives_to_install_dir)
+    cleanup_installation()
+
+
+def test_install_primitives_archive(primitives_to_install_archive):
+    setup_installation()
+    do_installation(primitives_to_install_archive)
+    cleanup_installation()
+
+
+def test_install_primitives_s3(primitives_to_install_dir, primitives_to_install_archive):
+    setup_installation()
+    s3_archive = "s3://featuretools-static/primitives_to_install.tar.gz"
+    do_installation(s3_archive)
+    cleanup_installation()
+
+
+def test_install_primitives_https(primitives_to_install_dir, primitives_to_install_archive):
+    setup_installation()
+    https_archive = "https://s3.amazonaws.com/featuretools-static/primitives_to_install.tar.gz"
+    do_installation(https_archive)
+    cleanup_installation()
+
+
+def test_list_primitive_files(primitives_to_install_dir):
+    files = list_primitive_files(primitives_to_install_dir)
+    custom_max_file = os.path.join(primitives_to_install_dir, "custom_max.py")
+    custom_mean_file = os.path.join(primitives_to_install_dir, "custom_mean.py")
+    custom_sum_file = os.path.join(primitives_to_install_dir, "custom_sum.py")
+    assert set(files) == {custom_max_file, custom_mean_file, custom_sum_file}
+
+
+def test_load_primitive_from_file(primitives_to_install_dir):
+    primitve_file = os.path.join(primitives_to_install_dir, "custom_max.py")
+    primitive_name, primitive_obj = load_primitive_from_file(primitve_file)
+    assert issubclass(primitive_obj, PrimitiveBase)
+
+
+def test_errors_more_than_one_primitive_in_file(bad_primitives_files_dir):
+    primitive_file = os.path.join(bad_primitives_files_dir, "multiple_primitives.py")
+    error_text = 'More than one primitive defined in file'
+    with pytest.raises(RuntimeError, match=error_text):
+        load_primitive_from_file(primitive_file)
+
+
+def test_errors_no_primitive_in_file(bad_primitives_files_dir):
+    primitive_file = os.path.join(bad_primitives_files_dir, "no_primitives.py")
```

Comment:
Should we test a ".py" that doesn't follow python syntax instead of an empty file?

Reply:
let's not worry about this case for now